test/python/test_rticonnextdds_discovery.py

```diff
@@ -0,0 +1,195 @@
+    xml_path = os.path.join(
+        os.path.dirname(os.path.realpath(__file__)),
+        "../xml/TestConnector.xml")
+    participant_profile="MyParticipantLibrary::DiscoveryTest"
+
+    with rti.open_connector(participant_profile, xml_path) as rti_connector:
+        yield rti_connector
+
+  @pytest.fixture(scope="function")
+  def discovery_reader_only_connector(self):
+    xml_path = os.path.join(
+        os.path.dirname(os.path.realpath(__file__)),
+        "../xml/TestConnector.xml")
+    participant_profile="MyParticipantLibrary::DiscoveryTestReaderOnly"
+
+    with rti.open_connector(participant_profile, xml_path) as rti_connector:
+        yield rti_connector
+
+  @pytest.fixture(scope="function")
+  def discovery_writer_only_connector(self):
+    xml_path = os.path.join(
+        os.path.dirname(os.path.realpath(__file__)),
+        "../xml/TestConnector.xml")
+    participant_profile="MyParticipantLibrary::DiscoveryTestWriterOnly"
+
+    with rti.open_connector(participant_profile, xml_path) as rti_connector:
+        yield rti_connector
+
+  @pytest.fixture(scope="function")
+  def discovery_reader_only_input(self, discovery_reader_only_connector):
+    return discovery_reader_only_connector.get_input("TestSubscriber::TestReader")
+
+  @pytest.fixture(scope="function")
+  def discovery_writer_only_output(self, discovery_writer_only_connector):
+    return discovery_writer_only_connector.get_output("TestPublisher::TestWriter")
+
+  def test_no_matches_input(self, discovery_reader_only_input):
+    # We should not match with any outputs at this point
+    matches = discovery_reader_only_input.get_matched_outputs()
+    assert matches is None
+
+    # We should timeout if we attempt to wait for a match
+    with pytest.raises(rti.TimeoutError) as excinfo:
+      change_in_matches = discovery_reader_only_input.wait_for_match(1)
+      assert change_in_matches == 0
+
+  def test_no_matches_output(self, discovery_writer_only_output):
+    # We should not match with any outputs at this point
+    matches = discovery_writer_only_output.get_matched_inputs()
+    assert matches is None
+
+    # We should timeout if we attempt to wait for a match
+    with pytest.raises(rti.TimeoutError) as excinfo:
+      change_in_matches = discovery_writer_only_output.wait_for_match(1)
+      assert change_in_matches == 0
+
+  def test_simple_matching(self, discovery_connector):
+    the_input = discovery_connector.get_input("MySubscriber::MyReader")
+    the_output = discovery_connector.get_output("MyPublisher::MyWriter")
+
+    # Both the input and output should match each other (and nothing else)
+    change_in_matches = the_input.wait_for_match(2000)
+    matches = the_input.get_matched_outputs()
+    assert change_in_matches == 1
+    assert isinstance(matches, list)
+    assert {'name': 'MyWriter'} in matches
+    change_in_matches = the_output.wait_for_match(2000)
+    matches = the_output.get_matched_inputs()
+    assert change_in_matches == 1
```

Comment:
why is this 1? You just called wait on line 86. The second call to wait without any new subscription should keep waiting.

Reply:
The second call to wait is on an output so the 1 is referring to the matched input in this case